### PR TITLE
Fix SMIL regression

### PR DESCRIPTION
**Pull request by mcginty**
SMIL parts weren't being sent, was a regression from my thumbnail change. Caught it after checking the differences in HTTP requests between us and other messengers. This fixes the AT&T content adaptation issue, was able to reproduce and then confirm fix.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/mms/PartParser.java [hunk 1]
```diff
@@ -67,9 +66,15 @@ public static int getSupportedMediaPartCount(PduBody body) {
...
+
     return ContentType.isImageType(contentType) ||
            ContentType.isAudioType(contentType) ||
            ContentType.isVideoType(contentType);
   }
+
+  private static boolean isNonTextMedia(PduPart part) {
+    return !ContentType.isTextType(Util.toIsoString(part.getContentType()));
+  }
 }
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/mms/PartParser.java]: Is there a reason to have both of these?

File: src/org/thoughtcrime/securesms/mms/ImageSlide.java [hunk 2]
```diff
@@ -207,7 +207,7 @@ public SMILRegionElement getSmilRegion(SMILDocument document) {
 
   @Override
   public SMILMediaElement getMediaElement(SMILDocument document) {
-    return SmilUtil.createMediaElement("img", document, new String(getPart().getName()));
+    return SmilUtil.createMediaElement("img", document, new String(getPart().getName()) + ".jpg");
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/mms/ImageSlide.java]: why .jpg? what if it's not a jpeg?

**Comment by mcginty** [comment]
Updated, all SMIL logic only resides in MMS send job now. Cleaner this way, and less code.

File: src/org/thoughtcrime/securesms/mms/PartParser.java [hunk 3]
```diff
@@ -57,9 +58,7 @@ public static int getSupportedMediaPartCount(PduBody body) {
     int partCount = 0;
 
     for (int i=0;i<body.getPartsNum();i++) {
-      String contentType = Util.toIsoString(body.getPart(i).getContentType());
-
-      if (isDisplayableMedia(contentType)) {
+      if (!isText(body.getPart(i))) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/mms/PartParser.java]: did you mean to leave these changes?

Comment (mcginty) [inline src/org/thoughtcrime/securesms/mms/PartParser.java]: moved back to isDisplayableMedia

File: src/org/thoughtcrime/securesms/jobs/MmsSendJob.java [hunk 4]
```diff
@@ -104,6 +105,7 @@ public MmsSendResult deliver(MasterSecret masterSecret, SendReq message)
     MmsRadio radio = MmsRadio.getInstance(context);
 
     try {
+      prepareMessageMedia(masterSecret, message, MediaConstraints.MMS_CONSTRAINTS, true);
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/jobs/MmsSendJob.java]: All this mutable object stuff keeps metastasizing.  Can we switch this to immutable?

Comment (mcginty) [inline src/org/thoughtcrime/securesms/jobs/MmsSendJob.java]: already done in the lollipop mms changes, which i'm working on now

Comment (moxie0) [inline src/org/thoughtcrime/securesms/jobs/MmsSendJob.java]: is this whole thing a part of that, or separate?

Comment (mcginty) [inline src/org/thoughtcrime/securesms/jobs/MmsSendJob.java]: separate

File: src/org/thoughtcrime/securesms/jobs/SendJob.java [hunk 5]
```diff
@@ -53,7 +53,7 @@ protected void prepareMessageMedia(MasterSecret masterSecret, SendReq message,
     }
   }
 
-  private void preparePart(MasterSecret masterSecret, MediaConstraints constraints,
+  protected void preparePart(MasterSecret masterSecret, MediaConstraints constraints,
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/jobs/SendJob.java]: Why is this protected?

**Pull request closed without merging**